src/main.rs

```diff
@@ -25,7 +25,8 @@ fn main() {
         )
         .unwrap();
 
-    let line_launcher = LineLauncher::default();
+    let mut line_launcher = LineLauncher::default();
+    line_launcher.progression = Progression::parse("C C C C Eb Eb Eb Eb").unwrap();
```

Comment:
The concept of a "default" for `Progression` + using that for a default `LineLauncher` (and then here manually mutating its `progression`) is probably feeling weird vs just forcing you to use eg `LineLauncher::new()`?

Reply:
Yeah, manually setting the field rather than passing that in in the constructor seems strange. (If there does end up being a reason for changing the progressing while `line_launcher` is running, I'd expect that it'll be more complicated than setting a field.)

Reply:
Ok updated to get rid of `LineLauncher::default()` + `Progression::default()` and implemented `From<Progression>` on `LineLauncher` and used that for creating our `LineLauncher` instance here (via `.into()`)